### PR TITLE
[Bot] Implement from issue #131: Implement core math utility functions

### LINKED ISSUE #131: Implement core math utility functions
Create a math utility module with functions to add and multiply two numbers, including basic input validation. This task is critical as it lays the foundation for the utility. Expected outputs are a fully functional `math_utils.py` file with complete implementations of the add and multiply functions.

**Acceptance Criteria**:
- math_utils.py is created and syntactically valid
- All imports resolve correctly to existing modules
- Functions add() and multiply() are fully implemented with input validation
- Code can be imported and executed without errors

**Files to modify**:
- `math_utils.py`

**Estimated effort**: 4 hours

**Parent**: #129

### PR DESCRIPTION
**Pull request by morotrade**
Auto-generated implementation.

Closes #131

Project root: `projects`

### CODE REVIEW

**Comment by morotrade** [comment]
### 🤖 AI Code Review



**Last Updated**: 2025-09-16 19:34:50 UTC

#### 📂 Project Root
`projects`

#### 📊 Summary
The implementation of core math utility functions is a good start, but it requires additional validation for edge cases and improvements in documentation and example usage.

#### 🎯 Issue Counts
- **🚫 BLOCKER**: 0
- **⚠️ IMPORTANT**: 1  
- **💡 SUGGESTION**: 2

#### 🔍 Detailed Findings

#### ⚠️ IMPORTANT
- **`projects/math_utils.py:12`**: The functions do not handle edge cases such as adding or multiplying non-numeric types like lists or strings.
  *Suggestion*: Consider adding type checks or extending the validation to handle more edge cases.

#### 💡 SUGGESTION
- **`projects/math_utils.py`**: The example usage in the main block could be improved with more diverse test cases.
  *Suggestion*: Add more examples to demonstrate the functionality, including edge cases.
- **`projects/math_utils.py`**: Consider adding a docstring for the module to describe its purpose.
  *Suggestion*: Add a module-level docstring at the top of the file.


#### 🔧 Suggested Patches
The following unified diffs stay under the project root and can be applied by the developer:

**Patch 1**

```diff
--- a/projects/math_utils.py
+++ b/projects/math_utils.py
@@ -1,6 +1,8 @@
+"""Math utility functions for basic operations."""
+
 def add(a, b):
     """Return the sum of a and b, ensuring both are numbers."""
     if not (isinstance(a, (int, float)) and isinstance(b, (int, float))):
         raise ValueError("Both arguments must be numbers.")
     return a + b
 
 def multiply(a, b):
     """Return the product of a and b, ensuring both are numbers."""
     if not (isinstance(a, (int, float)) and isinstance(b, (int, float))):
         raise ValueError("Both arguments must be numbers.")
     return a * b
 
 if __name__ == "__main__":
     # Example usage
-    print("Add 2 and 3:", add(2, 3))
-    print("Multiply 2 and 3:", multiply(2, 3))
+    print("Add 2 and 3:", add(2, 3))
+    print("Add 'a' and 3:", add('a', 3))  # Example of invalid input
+    print("Multiply 2 and 3:", multiply(2, 3))
+    print("Multiply 2 and 'b':", multiply(2, 'b'))  # Example of invalid input
```

---
> 🔄 **Auto-Review Loop**: This comment updates automatically when you push changes to this branch.  
> 🏷️ **Labels**: `need-fix` = blockers to resolve, `ready-to-merge` = all clear!

**Comment by morotrade** [comment]
🔎 Using project root: `projects` for PR-fix
